### PR TITLE
Centers map at user's location, if possible

### PR DESCRIPTION
**Pull request by a-stepin**
Add functionality to center map at the user's location if possible, and default coordinates if not possible.

When the user enters a map, they will be prompted to share their location. In the case that the user allows, the map centers around the user's current GPS coordinates. In all other cases (including navigator.geolocation not being available on the user's device), the map will center around a pair of default coordinates.

Also, move instantiation of user id to a more global context to avoid re-creating it with every invocation of the function.

### CODE REVIEW

**Review by Felomeng — APPROVED** [review]

File: src/main/webapp/main.js [hunk 1]
```diff
@@ -58,14 +62,43 @@ Promise.all([mapPromise, domPromise, firebasePromise]).then(() => {
...
+ * Returns the user's coordinates, if possible
+ * @returns{!Object} Tuple representing the user's latitude and longitude
+ */
+function getCoords(){
+
+  return new Promise(function(resolve, reject){
+    
+    if(navigator.geolocation){
+      navigator.geolocation.getCurrentPosition(function(position){
+        if(position.coords.latitude !== undefined){
```
Comment (AliceVlasov) [inline src/main/webapp/main.js]: nit: `if(position.coords.latitude)` is equivalent to `if(position.coords.latitude !== undefined)`

**Review by AliceVlasov — APPROVED** [review]

File: src/main/webapp/main.js (hunk 1 above)
Comment (a-stepin) [reply src/main/webapp/main.js]: Gotcha! I'll make that fix. Thanks for catching that!

File: src/main/webapp/chat.js [hunk 2]
```diff
@@ -127,7 +127,6 @@ function handleChatMessage(obj) {
   var node = document.createElement("div");
   var textnode = document.createTextNode(obj.name + ": " + obj.message);
 
-  let userId = firebase.auth().currentUser.uid;
   if(obj.uid === userId) {
```
Comment (salehhamadeh) [inline src/main/webapp/chat.js:131]: nit: Add a comment saying that `userId` is defined and in main.js

File: src/main/webapp/main.js [hunk 3]
```diff
@@ -58,14 +62,43 @@ Promise.all([mapPromise, domPromise, firebasePromise]).then(() => {
...
+          myMap = new ChapMap([DEFAULT_LAT, DEFAULT_LNG]);
+        })
+       
         initChat();
       });
     });
 
+/**
+ * Returns the user's coordinates, if possible
+ * @returns{!Object} Tuple representing the user's latitude and longitude
```
Comment (salehhamadeh) [inline src/main/webapp/main.js]: This returns a Promise. Promise can be used as a JsDoc type.
```
@returns{!Promise<Array<Number>>} Promise for a tuple representing the user's latitude and longitude.
```

**Review by salehhamadeh — APPROVED** [review]

**Pull request merged**